pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -232,6 +250,160 @@ func getMachineDefaulterOperation(platformStatus *osconfigv1.PlatformStatus) mac
+	machinesetServiceReference := admissionregistrationv1.ServiceReference{
+		Namespace: defaultWebhookServiceNamespace,
+		Name:      webhookName,
+		Path:      pointer.StringPtr(DefaultMachineSetValidatingHookPath),
+	}
+	return admissionregistrationv1.ValidatingWebhook{
+		AdmissionReviewVersions: []string{"v1beta1"},
+		Name:                    "validation.machineset.machine.openshift.io",
+		FailurePolicy:           &webhookFailurePolicy,
+		SideEffects:             &webhookSideEffects,
+		ClientConfig: admissionregistrationv1.WebhookClientConfig{
+			Service: &machinesetServiceReference,
+		},
+		Rules: []admissionregistrationv1.RuleWithOperations{
+			{
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{machine.GroupName},
+					APIVersions: []string{SchemeGroupVersion.Version},
+					Resources:   []string{"machinesets"},
+				},
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+					admissionregistrationv1.Update,
+				},
+			},
+		},
+	}
+}
+
+// NewMutatingWebhookConfiguration creates empty mutating webhook configuration
+func NewMutatingWebhookConfiguration() *admissionregistrationv1.MutatingWebhookConfiguration {
+	return &admissionregistrationv1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookConfigurationName,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+		Webhooks: []admissionregistrationv1.MutatingWebhook{
+			MachineMutatingWebhook(),
+			MachineSetMutatingWebhook(),
+		},
+	}
+}
+
+// MachineMutatingWebhook returns mutating webhooks for machine to apply in configuration
+func MachineMutatingWebhook() admissionregistrationv1.MutatingWebhook {
+	machineServiceReference := admissionregistrationv1.ServiceReference{
+		Namespace: defaultWebhookServiceNamespace,
+		Name:      webhookName,
+		Path:      pointer.StringPtr(DefaultMachineMutatingHookPath),
+	}
+	return admissionregistrationv1.MutatingWebhook{
+		AdmissionReviewVersions: []string{"v1beta1"},
+		Name:                    "default.machine.machine.openshift.io",
+		FailurePolicy:           &webhookFailurePolicy,
+		SideEffects:             &webhookSideEffects,
+		ClientConfig: admissionregistrationv1.WebhookClientConfig{
+			Service: &machineServiceReference,
+		},
+		Rules: []admissionregistrationv1.RuleWithOperations{
+			{
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{machine.GroupName},
+					APIVersions: []string{SchemeGroupVersion.Version},
+					Resources:   []string{"machines"},
+				},
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+				},
+			},
+		},
+	}
+}
+
+// MachineSetMutatingWebhook returns mutating webhook for machineSet to apply in configuration
+func MachineSetMutatingWebhook() admissionregistrationv1.MutatingWebhook {
+	machineSetServiceReference := admissionregistrationv1.ServiceReference{
```

Comment:
does this need to be exported?

Reply:
Yes, it is used in `pkg/operator/sync_test.go`: https://github.com/Danil-Grigorev/machine-api-operator/blob/9bce03bcac34eb91ef68609499c12b34ba7ca724/pkg/operator/sync_test.go#L253

Reply:
Thinking I can copy-paste existing ones in the test.

Reply:
no please, if it needs to be exported that's fine.